### PR TITLE
7 BOLA templates

### PR DESCRIPTION
**Pull request by arjun-akto**
7 templates

### CODE REVIEW

File: Broken-Object-Level-Authorization/BOLAJSONBodyParamArray.yml [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+  type: single
+  requests:
+    - req:
+        modify_body_param:
+          userKey: 
+            - "${random_ids}"
+validate:
+  response_code:
+    gte: 200
+    lt: 205
```
Comment (ankush-jain-akto) [inline Broken-Object-Level-Authorization/BOLAJSONBodyParamArray.yml:80]: 300?

File: Broken-Object-Level-Authorization/BOLAJSONBodyParamArray.yml [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    key: 
+      regex: "^UserId$|^UserID$|^User-ID$|^user_id$|^user-id$|^userid$|^username$|^Username$|^USERNAME$|^uId$|^uid$|^UID$|^member_id$|^MemberId$|^memberId$|^profile_id$|^ProfileId$|^profileId$|^userIdentifier$|^UserIdentifier$|^user_identifier$|^account_id$|^AccountID$|^account-id$|^account_id$|^customer_id$|^CustomerID$|^customer-id$|^customer_id$"
+    all_apis: true
+execute:
+  type: single
+  requests:
+    - req:
+        modify_body_param:
+          userKey: 
+            - "${random_ids}"
```
Comment (ankush-jain-akto) [inline Broken-Object-Level-Authorization/BOLAJSONBodyParamArray.yml:76]: should we handle query param here?

File: Broken-Object-Level-Authorization/BOLASystemResourceAccess.yml [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+        eq: Content-Type
+      value: 
+        extract: headerValue
+  or:
+    - request_payload:
+        for_one:
+          value:
+            contains_either:
+                - .jpg
+    - query_param:
```
Comment (ankush-jain-akto) [inline Broken-Object-Level-Authorization/BOLASystemResourceAccess.yml:65]: remove `query_param`

File: Broken-Object-Level-Authorization/BOLAwithFiles.yaml [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+  authenticated: true
+api_selection_filters:
+  response_code:
+    gte: 200
+    lt: 300
+  or:
+    - request_payload:
+        for_one:
+          key:
+            regex: file|document|attachment|saveAsName|File|Document|Attachment
```
Comment (ankush-jain-akto) [inline Broken-Object-Level-Authorization/BOLAwithFiles.yaml:58]: add `path` also?

File: Broken-Object-Level-Authorization/BOLAwithFiles.yaml [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+  response_code:
+    gte: 200
+    lt: 205
+  response_payload:
+    length:
+      gt: 0
+    percentage_match:
+      lt: 10
+    percentage_match_schema:
+      gte: 90
```
Comment (ankush-jain-akto) [inline Broken-Object-Level-Authorization/BOLAwithFiles.yaml:89]: Add

```
  response_payload:
    length:
      gt: 0
    not_contains_either:
      - Access Denied
      - Error
      - Unauthorized
      - Forbidden
      - Invalid
      - Token Expired
      - Not found
      - Bad Request
      - Internal server error
      - Service Unavailable
      - Rate limit exceeded

```

**Review by ayushaga14 — APPROVED** [review]

**Pull request merged**